### ISSUE TITLE
Incorrect OpenAPI 3 to Swagger 2 Conversion with Enums

### ISSUE THREAD

**Issue by trilom** [LucyBot-Inc/api-spec-converter#164]
I am attempting to use an Enum as described here
https://swagger.io/docs/specification/data-models/enums/ for OAS3.  (Here is the same page for OAS2 https://swagger.io/docs/specification/2-0/enums/) and store it in #/components/schemas for reuse.

Given the following sample OAS3 template:
```
openapi: 3.0.0
info:
  version: 0.0.1
  title: testsickle
paths:
  /user/account/{accountLevel}:
    post:
      parameters:
        - in: path
          name: accountLevel
          required: true
          schema:
            $ref: "#/components/schemas/accountLevel"
      responses:
        200:
          $ref: "#/components/responses/defaultResponse"
# complex objects have schema definitions
components:
  responses:
    defaultResponse:
      description: Success
      content:
        application/json: 
          schema:
            $ref: '#/components/schemas/DefaultResponseObject'
  schemas:
    accountLevel:
      type: string
      enum:
        - one
        - two
        - three
        - four
    DefaultResponseObject:
      required:
        - message
      properties:
        message:
          type: string
```

This validates fine.  However, after running the following command to convert:

```bash
api-spec-converter swagger/test.yml --from=openapi_3 --to=swagger_2 --syntax=yaml -c > swagger/test2.yml

Errors:

[
  {
    "code": "ONE_OF_MISSING",
    "params": [],
    "message": "Not a valid parameter definition",
    "path": [
      "paths",
      "/user/account/{accountLevel}",
      "post",
      "parameters",
      "0"
    ],
    "schemaId": "http://swagger.io/v2/schema.json#",
    "inner": [
      {
        "code": "ONE_OF_MISSING",
        "params": [],
        "message": "Data does not match any schemas from 'oneOf'",
        "path": [
          "paths",
          "/user/account/{accountLevel}",
          "post",
          "parameters",
          "0"
        ],
        "inner": [
          {
            "code": "OBJECT_MISSING_REQUIRED_PROPERTY",
            "params": [
              "schema"
            ],
            "message": "Missing required property: schema",
            "path": []
          },
          {
            "code": "OBJECT_MISSING_REQUIRED_PROPERTY",
            "params": [
              "type"
            ],
            "message": "Missing required property: type",
            "path": []
          }
        ]
      },
      {
        "code": "OBJECT_MISSING_REQUIRED_PROPERTY",
        "params": [
          "$ref"
        ],
        "message": "Missing required property: $ref",
        "path": [
          "paths",
          "/user/account/{accountLevel}",
          "post",
          "parameters",
          "0"
        ]
      }
    ]
  }
]
```
The resulting Swagger 2 template is as follows:
```
swagger: '2.0'
info:
  title: testsickle
  version: 0.0.1
paths:
  '/user/account/{accountLevel}':
    post:
      produces:
        - application/json
      parameters:
        - in: path
          name: accountLevel
          required: true
      responses:
        '200':
          description: Success
          schema:
            $ref: '#/definitions/DefaultResponseObject'
definitions:
  DefaultResponseObject:
    properties:
      message:
        type: string
    required:
      - message
  accountLevel:
    enum:
      - one
      - two
      - three
      - four
    type: string
x-components:
  responses:
    defaultResponse:
      description: Success
      schema:
        $ref: '#/definitions/DefaultResponseObject'
```
In the _parameters_ portion of the _/user/account/{accountLevel}_ path you can see that the enum is not correctly resolved.  I would expect the parameters object too look like this:
```
parameters:
  - in: path
    name: accountLevel
    required: true
    enum:
      - one
      - two
      - three
      - four
    type: string
```

Here is an example of my original test case.  The above is reduced down, but this is where I first discovered the error:
```
openapi: 3.0.0
info:
  version: 0.0.1
  title: testsickle
paths:
  /user/account/{accountLevel}:
    post:
      parameters:
        - $ref: "#/components/parameters/AccountLevel"
      responses:
        200:
          $ref: "#/components/responses/defaultResponse"
# complex objects have schema definitions
components:
  responses:
    defaultResponse:
      description: Success
      content:
        application/json: 
          schema:
            $ref: '#/components/schemas/DefaultResponseObject'
  parameters:
    AccountLevel:
      in: path
      name: accountLevel
      required: true
      schema:
        $ref: "#/components/schemas/accountLevel"
  schemas:
    accountLevel:
      type: string
      enum:
        - one
        - two
        - three
        - four
    DefaultResponseObject:
      required:
        - message
      properties:
        message:
          type: string
```

**Comment by bobby-brennan** [comment]
Thanks for the detailed report and minimal repro example. Looks like we weren't resolving the `$ref` in parameter schemas. Fixed now, and added your example as a test case.

The fix is available in 2.7.8 - let me know if you have any further issues.

**Comment by trilom** [comment]
🤙arigatou for the quick response 🙏

In the complex example above, this is the result:

```bash
api-spec-converter swagger/test.yml --from=openapi_3 --to=swagger_2 --syntax=yaml -c > swagger/test2.yml

Warnings:

[
  {
    "code": "UNUSED_DEFINITION",
    "message": "Definition is not used: #/definitions/accountLevel",
    "path": [
      "definitions",
      "accountLevel"
    ]
  }
]
```
Here is the resulting swagger 2.0 file.  As you can see, some of the information is duplicated.  It doesn't really bug me in my case, but it does produce some level of warning that might be an issue.
```
swagger: '2.0'
info:
  title: testsickle
  version: 0.0.1
paths:
  '/user/account/{accountLevel}':
    post:
      produces:
        - application/json
      parameters:
        - enum:
            - one
            - two
            - three
            - four
          in: path
          name: accountLevel
          required: true
          type: string
      responses:
        '200':
          description: Success
          schema:
            $ref: '#/definitions/DefaultResponseObject'
definitions:
  DefaultResponseObject:
    properties:
      message:
        type: string
    required:
      - message
  accountLevel:
    enum:
      - one
      - two
      - three
      - four
    type: string
x-components:
  parameters:
    AccountLevel:
      enum:
        - one
        - two
        - three
        - four
      in: path
      name: accountLevel
      required: true
      type: string
  responses:
    defaultResponse:
      description: Success
      schema:
        $ref: '#/definitions/DefaultResponseObject'
```
In a more complicated example, I get an output of a valid Swagger 2.0 file, and the OAS 3 file is valid (as per the Swagger Editor linting) however, with the api-spec-converter it is claming the following on the 2.7.8 version.

```bash
api-spec-converter swagger/swagger.yml --from=openapi_3 --to=swagger_2 --syntax=yaml -c > swagger/swagger2.yml

Errors:

[
  {
    "code": "MISSING_PATH_PARAMETER_DECLARATION",
    "message": "Path parameter is defined but is not declared: userStatus",
    "path": [
      "paths",
      "/user",
      "get",
      "parameters",
      "5"
    ]
  },
  {
    "code": "MISSING_PATH_PARAMETER_DECLARATION",
    "message": "Path parameter is defined but is not declared: accountLevel",
    "path": [
      "paths",
      "/user",
      "get",
      "parameters",
      "6"
    ]
  }
]
```
